test/dataframerow.jl

```diff
@@ -1,228 +1,265 @@
-r.c = "B"
-@test df.c[2] == "B"
-@test_throws BoundsError copy(r[[true, false, true]])
-
-r = DataFrameRow(sdf, 2, [3, 1])
-@test r[:] == r
-@test view(r, :) === r
-@test r[2] == "C"
-@test_throws BoundsError r[4]
-@test view(r, 2)[] == "C"
-view(r, 2)[] = "CC"
-@test df.c[3] == "CC"
-@test_throws MethodError r[true]
-@test_throws MethodError view(r, true)
-@test copy(r[[:c,:b]]) == (c = "CC", b = 1.2)
-@test copy(view(r, [:c,:b])) == (c = "CC", b = 1.2)
-@test copy(view(r, [:c,:b])) == (c = "CC", b = 1.2)
-@test copy(r[[false, true]]) == (c = "CC",)
-r.c = "C"
-@test df.c[3] == "C"
-
-#
-# Equality
-#
-@test DataFrameRow(df, 1, :) != DataFrameRow(df2, 1, :)
-@test DataFrameRow(df, 1, [:a]) == DataFrameRow(df2, 1, :)
-@test DataFrameRow(df, 1, [:a]) == DataFrameRow(df2, big(1), :)
-@test DataFrameRow(df, 1, :) != DataFrameRow(df, 2, :)
-@test DataFrameRow(df, 1, :) != DataFrameRow(df, 3, :)
-@test DataFrameRow(df, 1, :) == DataFrameRow(df, 4, :)
-@test ismissing(DataFrameRow(df, 2, :) == DataFrameRow(df, 5, :))
-@test isequal(DataFrameRow(df, 2, :), DataFrameRow(df, 5, :))
-@test ismissing(DataFrameRow(df, 2, :) != DataFrameRow(df, 6, :))
-@test !isequal(DataFrameRow(df, 2, :), DataFrameRow(df, 6, :))
-
-# isless()
-df4 = DataFrame(a=[1, 1, 2, 2, 2, 2, missing, missing],
-                b=Union{Float64, Missing}[2.0, 3.0, 1.0, 2.0, 2.0, 2.0, 2.0, 3.0],
-                c=[:B, missing, :A, :C, :D, :D, :A, :A])
-@test isless(DataFrameRow(df4, 1, :), DataFrameRow(df4, 2, :))
-@test !isless(DataFrameRow(df4, 2, :), DataFrameRow(df4, 1, :))
-@test !isless(DataFrameRow(df4, 1, :), DataFrameRow(df4, 1, :))
-@test isless(DataFrameRow(df4, 1, :), DataFrameRow(df4, 3, :))
-@test !isless(DataFrameRow(df4, 3, :), DataFrameRow(df4, 1, :))
-@test isless(DataFrameRow(df4, 3, :), DataFrameRow(df4, 4, :))
-@test !isless(DataFrameRow(df4, 4, :), DataFrameRow(df4, 3, :))
-@test isless(DataFrameRow(df4, 4, :), DataFrameRow(df4, 5, :))
-@test !isless(DataFrameRow(df4, 5, :), DataFrameRow(df4, 4, :))
-@test !isless(DataFrameRow(df4, 6, :), DataFrameRow(df4, 5, :))
-@test !isless(DataFrameRow(df4, 5, :), DataFrameRow(df4, 6, :))
-@test isless(DataFrameRow(df4, 7, :), DataFrameRow(df4, 8, :))
-@test !isless(DataFrameRow(df4, 8, :), DataFrameRow(df4, 7, :))
-
-# hashing
-@test hash(DataFrameRow(df, 1, :)) != hash(DataFrameRow(df, 2, :))
-@test hash(DataFrameRow(df, 1, :)) != hash(DataFrameRow(df, 3, :))
-@test hash(DataFrameRow(df, 1, :)) == hash(DataFrameRow(df, 4, :))
-@test hash(DataFrameRow(df, 2, :)) == hash(DataFrameRow(df, 5, :))
-@test hash(DataFrameRow(df, 2, :)) != hash(DataFrameRow(df, 6, :))
-
-# check that hashrows() function generates the same hashes as DataFrameRow
-df_rowhashes, _ = DataFrames.hashrows(Tuple(columns(df)), false)
-@test df_rowhashes == [hash(dr) for dr in eachrow(df)]
-
-# test incompatible frames
-@test_throws UndefVarError DataFrameRow(df, 1, :) == DataFrameRow(df3, 1, :)
-
-# test RowGroupDict
-N = 20
-d1 = rand(map(Int64, 1:2), N)
```

Comment:
I have rewritten the definition of `df5` because here is was not reproducible and it was possible that `d1` actually would hold only `1` or only `2` when the tests would fail.